Recurly/BaseClient.cs

```diff
@@ -106,6 +107,11 @@ private RestRequest BuildRequest(Method method, string url, Request body = null,
             var request = new RestRequest(url, method);
             request.JsonSerializer = Recurly.JsonSerializer.Default;
 
+            if (options != null)
+            {
+                request.AddHeaders(options.Headers);
```

Comment:
How does this work? I'm assuming it preserves the current headers or else it wouldn't work. Does it override existing headers?

Reply:
It's a [convenience method on RestClient](https://github.com/restsharp/RestSharp/blob/master/src/RestSharp/RestRequest.cs#L396) that basically just calls `AddHeader` for each item in the collection. Yes, it does preserve existing headers.